mscore/exportxml.cpp

```diff
@@ -4227,6 +4227,29 @@ static const FretDiagram* findFretDiagram(int strack, int etrack, int track, Seg
       return 0;
       }
 
+//---------------------------------------------------------
+//  commonAnnotations
+//---------------------------------------------------------
+
+static bool commonAnnotations(ExportMusicXml* exp, const Element* e, int sstaff)
+      {
+      if (e->isSymbol())
+            exp->symbol(toSymbol(e), sstaff);
+      else if (e->isTempoText())
+            exp->tempoText(toTempoText(e), sstaff);
+      else if (e->isStaffText() || e->isSystemText() || e->isText() || e->isInstrumentChange())
+            //exp->words(toText(e), sstaff); TODO
+            exp->words(static_cast<const Text*>(e), sstaff);
```

Comment:
Actually this should be wrong, `StaffText`, `SystemText` and `InstrumentChange` do not inherit from `Text` currently but from `TextBase`. This type cast will be wrong for these types.

Reply:
You could be right, I believe this causes toText() to fail, but unfortunately I am unable to investigate on short notice. Note that the cast was already there, I did not change it. Original code was:
                              case ElementType::STAFF_TEXT:
	                      case ElementType::TEXT:
	                      case ElementType::INSTRUMENT_CHANGE:
	                                    exp->words(static_cast<const Text*>(e), sstaff);

Reply:
Yes, that code could exist before introducing the new text classes hierarchy (which happened last summer, see https://github.com/musescore/MuseScore/commit/10ebbb9080d69bb3629ca48dfb76eb2a5aadef9e). It may happen not to cause any problems if nothing specific to `Text` class is used there but it is still better to correct this to prevent possible errors from happening.

Reply:
Indeed. This implies it should have been fixed when introducing the new text classes, but wasn't and did not cause issues for about a year. I am unable to do something about it for a few weeks. You can either merge anyway (and submit an issue to fix the problem later) or wait for me (or someone else) to fix it.

Reply:
OK, I'll merge it then and correct it with another commit, this should be a trivial change.

Reply:
See https://github.com/musescore/MuseScore/commit/1e5c2c365597d24054d332da73448940b1b13eaa.